### PR TITLE
Pass redux flash message to dismiss fn

### LINKED ISSUE #515: FlashMessage onDismiss should receive message object (including id)
Without having access to the ID, we can't call `removeMessage(id)` for the specific flash message

### PR DESCRIPTION
**Pull request by chawes13**
Resolves #515 

When we removed the [`<FlashMessage />` redux-flash dependency](https://github.com/LaunchPadLab/lp-components/pull/446), we accidentally introduced a bug where the dismiss function no longer had access to the message object. This came up on MTC when we were trying to pass the `onDismiss` function as a prop for a specific message. https://github.com/LaunchPadLab/mtc-rsp-client/pull/87/files/8250a907e7e2113508a4f18c13c7359e4bd02937

## Author Checklist

- [x] Add unit test(s)
- [ ] Update version in package.json (see the [versioning guidelines](https://github.com/LaunchPadLab/opex-public/blob/main/gists/npm-package-guidelines.md#pull-requests-and-deployments))
- [x] Update documentation (if necessary)
- [ ] Add story to storybook (if necessary)
- [ ] Assign dev reviewer

### CODE REVIEW

**Comment by chawes13** [comment]
@bhennes2 bump

File: src/indicators/flash-message-container.js [hunk 1]
```diff
@@ -29,26 +29,34 @@ import FlashMessage from './flash-message'
...
+      {messagesToDisplay.map((message) => {
+        const _onDismiss = message.props.onDismiss || onDismiss
+        const onMessageDismiss = _onDismiss ? () => _onDismiss(message) : null
+        return (
+          <FlashMessage
+            key={message.id}
+            message={message}
+            isError={message.isError}
+            {...rest}
+            {...message.props}
```
Comment (bhennes2) [inline src/indicators/flash-message-container.js:53]: Just confirming this was intentional to put `...message.props` after the `...rest`?

Comment (bhennes2) [reply src/indicators/flash-message-container.js:53]: I see the spec covers this!

**Review by bhennes2 — APPROVED** [review]
LGTM! (answered one of my own questions)

**Pull request merged**